src/neoxp/TransactionExecutor.cs

```diff
@@ -480,6 +481,88 @@ public async Task TransferNFTAsync(string contract, string tokenId, string sende
+        }
+
+        public async Task UnregisterCandidateAsync(string account, string password)
+        {
+            if (!chainManager.TryGetSigningAccount(account, password, out var wallet, out var accountHash))
+            {
+                throw new Exception($"{account} account not found.");
+            }
+            var publicKey = wallet.GetAccount(accountHash).GetKey()?.PublicKey;
+            using var builder = new ScriptBuilder();
+            builder.EmitDynamicCall(NativeContract.NEO.Hash, "unregisterCandidate", publicKey);
+
+            var txHash = await expressNode.ExecuteAsync(wallet, accountHash, WitnessScope.CalledByEntry, builder.ToArray()).ConfigureAwait(false);
+            await writer.WriteTxHashAsync(txHash, $"Unregister Candidate", json).ConfigureAwait(false);
+        }
+
+        public async Task VoteAsync(string account, string? publicKey, string password)
+        {
+            if (!chainManager.TryGetSigningAccount(account, password, out var wallet, out var accountHash))
+            {
+                throw new Exception($"{account} account not found.");
+            }
+
+            using var builder = new ScriptBuilder();
+            if (!string.IsNullOrEmpty(publicKey))
+            {
+                if (!ECPoint.TryParse(publicKey, ECCurve.Secp256r1, out ECPoint point))
+                {
+                    throw new Exception($"PublicKey is not valid.");
+                }
+                builder.EmitDynamicCall(NativeContract.NEO.Hash, "vote", accountHash, point);
+            }
+            else
+            {
+                builder.EmitDynamicCall(NativeContract.NEO.Hash, "vote", accountHash, null);
+            }
+
+            var txHash = await expressNode.ExecuteAsync(wallet, accountHash, WitnessScope.CalledByEntry, builder.ToArray()).ConfigureAwait(false);
+            await writer.WriteTxHashAsync(txHash, $"Vote/Unvote", json).ConfigureAwait(false);
+        }
+
+        public async Task<List<string>> ListCandidatesAsync()
+        {
+            using var builder = new ScriptBuilder();
+            builder.EmitDynamicCall(NativeContract.NEO.Hash, "getCandidates");
+
+            var result = await expressNode.InvokeAsync(builder.ToArray()).ConfigureAwait(false);
+            var stack = result.Stack;
+            var list = new List<string>();
+            try
+            {
+                if (result.State != VMState.FAULT
+                        && result.Stack.Length >= 1
+                        && result.Stack[0] is Neo.VM.Types.Array array)
+                {
+                    for (var i = 0; i < array.Count; i++)
+                    {
+                        var value = (Neo.VM.Types.Array)array[i];
+                        Console.Write(((Neo.VM.Types.ByteString)value?[0])?.GetSpan().ToHexString() + "\t");
+                        Console.WriteLine(((Neo.VM.Types.Integer)value?[1]).GetInteger());
```

Comment:
should it be `writer`?

Reply:
> should it be `writer`?

It's same to neo-cli

Reply:
I understand that, but copy and pasting isn't your friend 99% of the time. Things can easily go overlooked. `neo-express` doesn't use `console` as writer for the most the stuff. Because it uses filesystem handlers and other custom handlers. To handle console writing. Please verify that is done correctly.